google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/BigtableInterceptorProvider.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.data.v2.stub;
+
+import com.google.api.core.BetaApi;
+import com.google.api.gax.grpc.GrpcInterceptorProvider;
+import com.google.cloud.bigtable.data.v2.stub.metrics.ClientHeaderInterceptor;
+import com.google.common.collect.ImmutableList;
+import io.grpc.ClientInterceptor;
+import java.util.List;
+
+@BetaApi
+public class BigtableInterceptorProvider implements GrpcInterceptorProvider {
```

Comment:
I think it would be better to name this FixedIterceptorProvider and put this in the gaxx package and then in the future to move it to gax proper. (This implies that the construction of the ClientHEaderInterceptor should move out of this class to the caller)